packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.base.tsx

```diff
@@ -21,12 +22,19 @@ export class CheckboxBase extends React.Component<ICheckboxProps, ICheckboxState
   private _classNames: { [key in keyof ICheckboxStyles]: string };
 
   public static getDerivedStateFromProps(props: ICheckboxProps, state: ICheckboxState): ICheckboxState {
+    if (!props.defaultIndeterminate && state.isIndeterminate) {
+      return {
+        ...state,
+        isIndeterminate: !!props.indeterminate
+      };
+    }
     if (props.checked !== undefined) {
       return {
         ...state,
```

Comment:
You can remove this spread too. And the last line of the method can return null.